### DISCUSSION TITLE
Why read_csv_auto adds leading zeros to column's values while importing my CSV file?

### DISCUSSION THREAD

**Discussion by parvizfarnia** [duckdb/duckdb#6999] [Q&A] [answered]
Hello everyone

I would like to ask a question about `read_csv_auto` function that I use to import CSV files.

Actually I see a strange behaviour of this function on one of my files. Here is a test case to reproduce the problem:

Add the following data (timestamp values) in a text file that we name: 
**data.txt**
```
23-04-07T16:47:10
23-04-07T16:47:11
23-04-07T16:47:12
23-04-07T16:47:13
23-04-07T16:47:14

```

Now let's create a table for the sake of our test:
`create table tmptab(dateval text);`

And finally let's import our file into this table
```
insert into tmptab(dateval) 
select 
    * 
from 
    read_csv_auto(
        'C:/tmp/logs/data.txt', 
        header = False, 
        delim = '|', 
        quote = '"', 
        nullstr = '', 
        names = ['dateval']
    );

```
Now if I check the imported data in my table, here is what I see
```
D select * from tmptab;
┌─────────────────────┐
│       dateval       │
│       varchar       │
├─────────────────────┤
│ 0023-04-07 16:47:10 │
│ 0023-04-07 16:47:11 │
│ 0023-04-07 16:47:12 │
│ 0023-04-07 16:47:13 │
│ 0023-04-07 16:47:14 │
└─────────────────────┘
D

```

I don't really understand why there are two leading zeros for every row, also the letter `'T'` of the timestamp has been disappeared from my values. It seems that DuckDB has somewhat interpreted my values in a particular way. 

How can I solve this problem and import my data as is?

Thanks in advance.

**Comment by Mytherin** [comment, marked as answer]
The value is interpreted as a timestamp, with `23` being the year (hence `0023`). You could specify a `VARCHAR` type for the column (as `types=['varchar']`) or specify `all_varchar=True` to read all columns as varchars.

**Reply by parvizfarnia** [reply]
Thank you very much, this solved the problem. 


```
insert into tmptab(dateval) 
select 
    * 
from 
    read_csv_auto(
        'C:/tmp/logs/data.txt', 
        header = False, 
        delim = '|', 
        quote = '"', 
        nullstr = '', 
        names = ['dateval'],
        types = ['varchar']
    );

D select * from tmptab;
┌───────────────────┐
│      dateval      │
│      varchar      │
├───────────────────┤
│ 23-04-07T16:47:10 │
│ 23-04-07T16:47:11 │
│ 23-04-07T16:47:12 │
│ 23-04-07T16:47:13 │
│ 23-04-07T16:47:14 │
└───────────────────┘
D
```

I hadn't pay attention to this option. Now I see that I have to add it in multiple programs as I have many columns defined as `TEXT` which according to [Data types](https://duckdb.org/docs/sql/data_types/overview.html) is an alias for `VARCHAR` data type.

Thank you very much for your time and your help.

Regards.
